packages/user-operation-controller/src/utils/validation.ts

```diff
@@ -0,0 +1,191 @@
+import { isStrictHexString } from '@metamask/utils';
+import type { Struct, StructError } from 'superstruct';
+import {
+  assert,
+  define,
+  func,
+  object,
+  optional,
+  refine,
+  string,
+} from 'superstruct';
+
+import { EMPTY_BYTES } from '../constants';
+import type {
+  PrepareUserOperationResponse,
+  SignUserOperationResponse,
+  UpdateUserOperationResponse,
+} from '../types';
+
+/**
+ * Validate a request to add a user operation.
+ * @param request - The request to validate.
+ */
+export function validateAddUserOperationRequest(request: any) {
```

Comment:
Just the first two methods since the `addUserOperation` method doesn't use types for either the request itself or the options, and we have no real need for a type here as we want to verify at runtime if any arbitrary data matches a given schema.